### PR TITLE
collect : new gui for numeric and date/time values

### PR DESCRIPTION
**Pull request by AlicVB**
concern day/time/aperture/iso/focal-length
the gui let the user choose the comparison operator and set the values (2 for range op)
the entries to set values have autocompletion support with all the values previously shown as a list.

### CODE REVIEW

File: src/libs/collect.c [hunk 1]
```diff
@@ -715,6 +676,335 @@ static GtkTreeModel *_create_filtered_model(GtkTreeModel *model, dt_lib_collect_
...
+    guint y, m, dd;
+    gtk_calendar_get_date(calendar, &y, &m, &dd);
+
+    // we adjust the time part depending of the operator
+    char dtxt[40] = { 0 };
+    switch(dt_bauhaus_combobox_get(d->num_op))
+    {
+      case 0:
+      case 5:
+        sprintf(dtxt, "%04d:%02d:%02d%%", y, m, dd);
```
Comment (LebedevRI) [inline src/libs/collect.c]: could you please not use `sprintf()` (at all, if possible), but use `snprintf()`
some time ago i have converted all/most sprintf to snprintf..

File: src/libs/collect.c [hunk 2]
```diff
@@ -715,6 +676,335 @@ static GtkTreeModel *_create_filtered_model(GtkTreeModel *model, dt_lib_collect_
...
+  if(entry)
+  {
+    guint y, m, dd;
+    gtk_calendar_get_date(calendar, &y, &m, &dd);
+
+    // we adjust the time part depending of the operator
+    char dtxt[40] = { 0 };
+    switch(dt_bauhaus_combobox_get(d->num_op))
+    {
+      case 0:
```
Comment (LebedevRI) [inline src/libs/collect.c]: i feel like this 0..0 should be a enum

File: src/libs/collect.c (hunk 1 above)
Comment (AlicVB) [inline src/libs/collect.c]: fixed

**Comment by TurboGit** [comment]
Nice and working fine to me. The UI looks ok, I would probably put less space between the lines operator and value, but that's a very minor point.

**Comment by houz** [comment]
Could you please show a screenshot how it's supposed to look? For me it's widgets all over the place, strings that don't make sense (p.e. "inferior") and the 2nd entry "value" is confusing and lacks an obvious reason to be there.

**Comment by AlicVB** [comment]
it's supposed to look like that : img1 img2
For the strings, my English is pretty bad ;) How would you call them ? "lower", "lower than" ?

**Comment by houz** [comment]
And how does the other entry box fit into that scheme? To me that looks terrible. And the context menu …

**Comment by AlicVB** [comment]
well...
Main goal is to have a more easy way to deal with numeric values. imho numeric values are mainly useful if used with operators like >, < or [;]. So actually the steps are : 
- finding the right value in the list (can be quite tedious, as the list can be very very long for focal_length or dates)
- double-click to select it
- click on the right place inside the top entry box (tedious too, as it's tiny, and often content of the entry is larger than the physical size of the box, esp for dates)
- add the operator
(and here, I don't speak about setting a range operator for dates...)

**Comment by AlicVB** [comment]
oh... and the only context menu is the built-in gtk one (with copy/paste...)
So I guess you are speaking of the autocompletion ?
It's here to let user know what values are actually present in the db, like we do when displaying the list actually. I've just added a "calendar..." line for dates, so user can pick up a date without going thought all the list or typing manually...

**Comment by houz** [comment]
I don't question that selecting ranges is a good thing. But the UI you came up with just doesn't work. And I don't think it could be fixed by moving things around a bit.

**Comment by AlicVB** [comment]
ok, let's tag this as incomplete until we find a better way to display/select numeric/dates and ranges. That's the main point ;)

**Comment by AlicVB** [comment]
Following houz and hanatos comments about this ui, with messy entries, let's close this pr.

althought it doesn't resolve all, pr #1172 can be a solution to avoid the "infinite" list of dates.

**Pull request closed without merging**